src/internal_macros.rs

```diff
@@ -416,3 +433,24 @@ macro_rules! serde_struct_impl {
         }
     )
 }
+
+#[cfg(test)]
+#[cfg(feature = "serde")]
```

Comment:
Can you gate on the `serde-decimal` feature rather than directly on `serde`, since you're using multiple crates?